File: Data Structures & Algorithms/valid-sudoku/submission-0.java
class Solution {
     public boolean isValidSudoku(char[][] board) {
        HashSet<Character>[] rows = new HashSet[9];
        HashSet<Character>[] cols = new HashSet[9];
        HashSet<Character>[] squares = new HashSet[9];
        for (int i = 0; i < 9; i++) {
            rows[i] = new HashSet<>();
            cols[i] = new HashSet<>();
            squares[i] = new HashSet<>();
        }

        for (int i = 0; i < 9; i++) {
            for (int j = 0; j < 9; j++) {
                // Check the row
                char num = board[i][j];
                if (num != '.') {
                    // Check the row
                    if (rows[i].contains(num)) return false;
                    rows[i].add(num);

                    // Check the col
                    if (cols[j].contains(num)) return false;
                    cols[j].add(num);

                    // Check the square
                    int index = (i / 3) * 3 + (j / 3);
                    if (squares[index].contains(num)) return false;
                    squares[index].add(num);
                }
            }
        }
        return true;
    }
}

File: Data Structures & Algorithms/flood-fill/submission-0.cs
public class Solution {
    public int[][] FloodFill(int[][] image, int sr, int sc, int color) {
        HashSet<(int, int)> seen = new HashSet<(int, int)>();

        dfs(image, sr, sc, seen, color, image[sr][sc]);

        return image;
    }
    public void dfs(int[][] image, int row, int col, HashSet<(int, int)> seen, int color, int orig) {
        seen.Add((row, col));
        if (row + 1 < image.Length && !seen.Contains((row + 1, col))) {
            if (orig == image[row+1][col]) {
                dfs(image, row + 1, col, seen, color, orig);
            }
        }
        if (row - 1 > -1  && !seen.Contains((row - 1, col))) {
            if (orig == image[row - 1][col]) {
                dfs(image, row - 1, col, seen, color, orig);
            }
        }
        if (col + 1 < image[0].Length  && !seen.Contains((row, col + 1))) {
            if (orig == image[row][col + 1]) {
                dfs(image, row, col + 1, seen, color, orig);
            }
        }
        if (col - 1 > -1  && !seen.Contains((row, col - 1))) {
            if (orig == image[row][col - 1]) {
                dfs(image, row, col - 1, seen, color, orig);
            }
        }
        image[row][col] = color;
    }
}